pandas/tests/series/test_diff.py

```diff
@@ -0,0 +1,24 @@
+from pandas import (
+    Series
+)
+
+from numpy import nan
+
+
+def test_diff():
+    data = Series(
+        [0, -1, -2, -3, -4, -3, -2, -1, 0, -1, -1, 0, -1, -2, -3, -2, 0]
+    )
+    filtered = data.between(-2, 0, inclusive=True)
+    diff_boolean = filtered.diff()
```

Comment:
Can you remove this and just make the assert on the the result of `data.between`? I'm not sure why `diff` is needed.

Reply:
And use `tm.assert_series_equal` instead of `assert`

Reply:
Alright, first comment, release notes, do you have a link for how to write that?
Second comment, agreed, I'll shorten it.
Third comment, the diff is what I used to test the feature, it causes the issue in my test build (diff on boolean series). So if I remove it, the bug will not show up when testing.
Last comment, I tried using pytest and got some errors so went with the old fashioned way.

Reply:
Apparently the warning was added in numpy 1.13.0, 3 years ago.

Reply:
So is the issue with `Series.between` or `Series.diff`? Let's use one or the other to keep the test focused.

Reply:
Fixed